system/src/system_power_manager.cpp

```diff
@@ -165,6 +187,8 @@ void PowerManager::handleUpdate() {
           // so just check input current source register whenever we are in this state
           if (power.getInputCurrentLimit() != DEFAULT_INPUT_CURRENT_LIMIT) {
             power.setInputCurrentLimit(DEFAULT_INPUT_CURRENT_LIMIT);
+            power.setInputVoltageLimit(3880);
```

Comment:
Let's be cautious and enable this only if `HAL_INCREASE_CHARGING_CURRENT_WHEN_POWERED_BY_VIN` is `1`, so for Gen 3 only.